pkg/products/apicurioregistry/reconciler.go

```diff
@@ -0,0 +1,308 @@
+	"github.com/integr8ly/integreatly-operator/pkg/resources/events"
+	"github.com/integr8ly/integreatly-operator/pkg/resources/marketplace"
+	"github.com/integr8ly/integreatly-operator/pkg/resources/owner"
+	"github.com/integr8ly/integreatly-operator/version"
+	appsv1 "github.com/openshift/api/apps/v1"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	defaultInstallationNamespace = "apicurio-registry"
+	manifestPackage              = "integreatly-apicurio-registry"
+	replicas                     = 1
+	amqStreamsTopicPartitions    = 3
+	amqStreamsTopicReplicas      = 3
+	amqStreamsTopicCleanupPolicy = "compact"
+)
+
+// Reconciler reconciles everything needed to install Apicurio Registry. The resources that it works
+// with are considered secondary resources in the context of the installation controller.
+type Reconciler struct {
+	Config        *config.ApicurioRegistry
+	ConfigManager config.ConfigReadWriter
+	logger        *logrus.Entry
+	mpm           marketplace.MarketplaceInterface
+	*resources.Reconciler
+	recorder record.EventRecorder
+}
+
+// NewReconciler creates a new Reconciler.
+func NewReconciler(configManager config.ConfigReadWriter, installation *integreatlyv1alpha1.RHMI, mpm marketplace.MarketplaceInterface, recorder record.EventRecorder) (*Reconciler, error) {
+	config, err := configManager.ReadApicurioRegistry()
+	if err != nil {
+		return nil, fmt.Errorf("could not read apicurio registry config: %w", err)
+	}
+
+	if config.GetNamespace() == "" {
+		config.SetNamespace(installation.Spec.NamespacePrefix + defaultInstallationNamespace)
+	}
+	if config.GetOperatorNamespace() == "" {
+		if installation.Spec.OperatorsInProductNamespace {
+			config.SetOperatorNamespace(config.GetNamespace())
+		} else {
+			config.SetOperatorNamespace(config.GetNamespace() + "-operator")
+		}
+	}
+
+	logger := logrus.NewEntry(logrus.StandardLogger())
+
+	return &Reconciler{
+		Config:        config,
+		ConfigManager: configManager,
+		logger:        logger,
+		mpm:           mpm,
+		Reconciler:    resources.NewReconciler(mpm),
+		recorder:      recorder,
+	}, nil
+}
+
+// GetPreflightObject returns an object that will be checked in the preflight checks in the main
+// Installation controller to ensure there isn't a conflicting Camel K installation.
+func (r *Reconciler) GetPreflightObject(ns string) runtime.Object {
+	return &appsv1.DeploymentConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "apicurio-registry-operator",
+			Namespace: ns,
+		},
+	}
+}
+
+func (r *Reconciler) VerifyVersion(installation *integreatlyv1alpha1.RHMI) bool {
+	return version.VerifyProductAndOperatorVersion(
+		installation.Status.Stages[integreatlyv1alpha1.SolutionExplorerStage].Products[integreatlyv1alpha1.ProductApicurioRegistry],
```

Comment:
It's currently in the solution explorer stage because it requires AMQ Streams to be installed first. We can review that once we add support for PostgreSQL.

Reply:
Are we OK with this change?

Reply:
yes, this is fine